core/controller_test.go

```diff
@@ -48,16 +49,16 @@ var _ = Describe("Controller", func() {
 			Context(".Create", func() {
 				It("does not error on create with valid opts", func() {
 					fakeClient.CreateVolumeReturns(nil)
-					createRequest := &resources.CreateRequest{Name: "dockerVolume1", Opts: map[string]interface{}{"Filesystem": "gpfs1"}}
+					createRequest := &resources.CreateVolumeRequest{Name: "dockerVolume1", Backend:Backend, Opts: map[string]interface{}{"Filesystem": "gpfs1"}}
```

Comment:
any specific reason to change to returning objects instead of references ( pointers) ? . This is better memory and heap management. The HttpResponseWriter eventually dereferences and send the payload to docker .  Returning references should be sufficient in my opinion